pandas/core/categorical.py

```diff
@@ -330,9 +330,9 @@ def __init__(self, values, categories=None, ordered=False,
 
     def copy(self):
         """ Copy constructor. """
-        return Categorical(values=self._codes.copy(),
```

Comment:
maybe make a ``_constructor`` as that is the patter we are using for other classes